### ISSUE TITLE
train.py (Row 188 )  change    hyp['cls'] *= nc / 80  to hyp['cls'] *= nc / 6  ?

### ISSUE THREAD

**Issue by china56321** [ultralytics/yolov5#167]
1: Need i to change the  parameter in train.py (Row 188 )when i train my own datasets  having 6 classes  ?(Change 80 to  6  when train my own datasets  ?)
Row 188:
  # Model parameters
    hyp['cls'] *= nc / 80.  # scale coco-tuned hyp['cls'] to current dataset

2: where and How  to freeze some layer ?

3: Thanks..........................

**Comment by china56321** [comment]
> Need i to change the parameter in train.py (Row 188 )when i train my own datasets having 6 classes ?(Change 80 to 6 when train my own datasets ?)
> Row 188:
> 
> # Model parameters
> ```
> hyp['cls'] *= nc / 80.  # scale coco-tuned hyp['cls'] to current dataset
> ```

**Comment by glenn-jocher** [comment]
You should neither change anything on the line you reference nor freeze any layers if you want the best results.

**Comment by haobabuhaoba** [comment]
so，What did you do in the end？
你，最后是怎么处理的？

**Comment by glenn-jocher** [comment]
@haobabuhaoba the line is fine as-is, do not change it.

**Comment by snop222** [comment]
If my nc (number of classes) is very large, for example, 500 even 1000, still do not change it ?

**Comment by glenn-jocher** [comment]
@snop222 customizations on python files are up to you. For best results I would not modify any python files in this repo unless you 1) first establish a baseline result with the default code, and 2) know what you are doing well.

**Comment by snop222** [comment]
Thank you for answer. 
My dataset is 542 classes, 119,068 images, 362,968 boxes.
With the default code, I run it on 4-32G-GPU with:
`python -m torch.distributed.launch\
 --nproc_per_node 4\
 ./pys2/train.py\
 --weights ./weights/yolov5x.pt\
 --hyp ./data/hyp.finetune.yaml\
 --sync-bn\
 --device 0,1,2,3\
 --img-size 224\
 --batch-size 512\
 --epochs 2500`
The curves are:
http://www.snop.idv.tw/c1.jpg
http://www.snop.idv.tw/c2.jpg
http://www.snop.idv.tw/c3.jpg
http://www.snop.idv.tw/c4.jpg
The precision is decreasing from 0.99 !!

**Comment by glenn-jocher** [comment]
@snop222 your val objectness loss is overfitting.

You can ignore P and R, mAP is the only absolute metric that matters.

**Comment by snop222** [comment]
Thanks very much for your hint.
But, how to make the mAP to be the only absolute metric that matters ?
